[Toh Wang Bin] iP

trash-bin99 opened the pull request:
# Welcome to the world's most boring chatbot!!! 🥇 
<h3>

Brought to you by __*trash-bin99*__ 🗑️ , the world's most boring student 🙃

<h6> 

In fact, you might be better off watching [this](https://www.youtube.com/watch?v=dQw4w9WgXcQ) instead of reading this text.

<h4>

> How is this the __*most*__ boring chatbot?
* Essentially zero innovation 😪 
* Bad code that is easy to point at and ridicule 😂 
* Even the PR description is yawn-inducing

I can even list some things that you are likely to do while reviewing my code:
1. Sleep
2. Complain about how the code is written, then realise it gets worse
3. Start booting up Minecraft, so you can throw yourself in lava after reading my code
4. Reminiscing about the times when your brain hadn't read my code yet

Okay, that is probably enough self-depreciating humor.  

Here are some of the innovations that i have planned <sup><sub>(but have zero intentions of doing)</sup></sub>
- [ ] Give the chatbot a coherent **personality**
- [ ] Add stupid **easter eggs** that only I would know
- [ ] Further simplify the **code** <sub>(not exactly an innovation, but still)</sub>

> How fun is interacting with your chatbot? 

Well, let me sum it up with a simple function:

```
private void interactWithDuke() {
    while(true) {
        System.out.println("Welcome to purgatory, you can never escape.");
    }
}
```

Commit:
Add Gradle support

trash-bin99 committed:
Add Level-1 increment.

trash-bin99 committed:
Add Level-2 increment

trash-bin99 committed:
Add Level-3 increment

trash-bin99 committed:
Add increment Level-4 requirements

trash-bin99 committed:
Add sample testcases

trash-bin99 committed:
Simplify code of Tasks

trash-bin99 committed:
Fix a minor bug for Events

trash-bin99 committed:
Add checks for potential errors.

trash-bin99 committed:
Convert task array to ArrayList

trash-bin99 committed:
Add delete function for the chatbot

trash-bin99 committed:
Add some comments and improve chatbot

trash-bin99 committed:
Add Javadoc and comments

trash-bin99 committed:
Add data storing functionality to the chatbot

Incorporation of data storing capabilities to the chatbot
requires the addition of several new methods, as well as
a new class. We will store data using the character "|" as
a separatory device.

trash-bin99 committed:
Add further functionality to data saving capabilities

Add the ability to save data to the file on creating and editing
task files.

trash-bin99 committed:
Add Level-8 functionality

Improves date and time recognition by Task classes.

trash-bin99 committed:
Merge branch 'branch-level-7' into master

trash-bin99 committed:
Merge branch 'branch-level-8' into master

trash-bin99 committed:
Bugfix of DataParser class

Change of parameter types in parse method from String to LocalDate.

trash-bin99 committed:
Create Storage class to handle data loading and writing

The new Storage class contains methods and processes that run on
startup, for the handling of data.

trash-bin99 committed:
Create Storage class to handle data loading and writing.

The new Storage class contains methods and processes that run on
startup, for the handling of data.

trash-bin99 committed:
Merge branch 'master' of github.com:trash-bin99/ip

trash-bin99 committed:
Refactor code chunks into separate classes

Separate code in Duke.java to more relevant classes.

trash-bin99 committed:
Consolidate code in Parser type classes

trash-bin99 committed:
Split code into more classes

Creation of new Ui class to handle print messages. Deletion of
empty file DataParser.

trash-bin99 committed:
Move Duke files into a package

trash-bin99 committed:
Add Gradle tests for Duke

Add 2 simple tests that target the Storage and TaskList classes,
which subsequently also test the other classes.

trash-bin99 committed:
build.gradle: Edit mainClassName under application to facilitate
creation of fatJar file

Edit from "seedu.duke.Duke".

trash-bin99 committed:
Add Javadocs comments

trash-bin99 committed:
Tweak code to adhere to coding standards

trash-bin99 committed:
Add Level-9 increment

Add find method to search for a Task with a specified search term
inside.

trash-bin99 committed:
Merge branch 'branch-A-Javadoc' into branch-A-CodingStandard

trash-bin99 committed:
Merge branch 'branch-A-CodingStandard' into branch-Level-9

trash-bin99 committed:
Merge branch 'branch-Level-9'

trash-bin99 committed:
Duke.java : Rename booleans to comply with code standard

trash-bin99 committed:
Reorganise methods in classes

Moved and organised methods in classes, to make code more readable.

trash-bin99 committed:
Move enum Tasks into the Task class

trash-bin99 committed:
Rename class Message to Ui

driip-ddrop commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,170 @@
+        //case if task stored is a Event
+        case "E":
+            Task event = new Events(splitArray[2], Parser.parseDate(splitArray[3]));
+            if (splitArray[1].equals("1")) { event.setCompleted(); }
```
The "event.setCompleted();" should be on the next line. I noticed this in several other lines as well.

driip-ddrop commented on src/main/java/duke/TaskCreator.java:
```diff
@@ -0,0 +1,116 @@
+                    return;
+                }
+                break;
+        }
```
There should not be indentation case in switch statements.

driip-ddrop commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,94 @@
+    //The List containing the Tasks
+    private List<Task> taskList;
+    //An integer storing the number of tasks in the list
+    private int taskNumber;
```
The name taskNumber maybe misleading, perhaps a better name could be something like totalTasks?

driip-ddrop commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,94 @@
+     * Constructor for a TaskList instance. Initialises the taskList.
+     */
+    public TaskList() {
+        this.taskList = new ArrayList<>();
```
This is ok but I think that the "this" in "this.taskList" is not necessary.

driip-ddrop commented on src/test/java/duke/DukeTest.java:
```diff
@@ -0,0 +1,38 @@
+
+import org.junit.jupiter.api.Test;
+
+public class DukeTest {
```
Perhaps create new classes like TaskListTest to test for methods in TaskList rather than putting everything in DukeTest

driip-ddrop reviewed (commented):
Other than some very minor coding violations, I like that your code is clean and includes comments at most of the lines so that those reading through get an idea of what is going on.  :)) good job!!

trash-bin99 committed:
Edit code according to comments made by peer

Separated tests by classes they are testing, and corrected code
violations.

trash-bin99 committed:
Revise code again according to checkstyle

Add config files for checkstyle to operate, and further revise code
in accordance to checkstyle regulations.

trash-bin99 committed:
Implement all basic functionalities required for the chatbot GUI

Heavily referenced from
"https://se-education.org/guides/tutorials/javaFx.html".

trash-bin99 commented:
Hey @driip-ddrop, thanks for the review! I have made the necessary changes as suggested.

trash-bin99 committed:
Adapt code to incorporate GUI

Now, most methods that formerly printed strings out to the console
will instead return strings, which will be handed to the GUI
for output to the user.

trash-bin99 committed:
Update tests and JavaDocs

trash-bin99 committed:
Merge branch 'branch-level-10'

joenzkc commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,175 @@
+    private static String run(String input) {
+        //initialise required classes
+        //check if initialised:
+        if (!isInitialised) {
```
Good way of making the Happy path prominent! (Using guard clauses)

joenzkc commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,172 @@
+                deadline.setCompleted();
+            }
+            return deadline;
+        //no need for break, as function has terminated at the return statement
```
Nice explanation to explain to other programmers why you leave out the break in the case statement! 👍

joenzkc commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,91 @@
+    private String filePath;
+    //path of folder containing data file
+    private String directoryPath;
+    //FIle containing stored data
```
Is this a typo with FIle? But I like how you explain what each field represents!

joenzkc commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,185 @@
+    private static final String NO_TASK_ERROR = "There are currently no tasks! Stop being so lazy "
+            + "and start adding your tasks!\n";
+
+    private static final String TASK_ERROR = "That task does not exist! Don't try to trick me!\n";
```
Nice use of final variables for re usability across different methods!

joenzkc reviewed (commented):
Overall, coding standard was great! Use of OOP was also very good. Keep up the good work 😄

trash-bin99 commented:
Hi @joenzkimchan, thanks for taking the time to view my code, and leaving the kind feedback (as well as spotting my typo)!

trash-bin99 committed:
Bugfix for case when save file does not exist

Also, edited some comments and made minor edits to the
build.gradle file.

trash-bin99 committed:
Add assertions to the code

Code does not make use of assertions.

Adding assertions will help to avoid situations where a bug has
occurred but the code continues to run, potentially causing damage
to saved data.

Let's add assertions to various classes.

This will prevent damaging bugs from occurring.

We shall also make some edits to JavaDocs and use Platform.exit() to
exit from the GUI when the users wants to.

trash-bin99 committed:
Merge pull request #1 from trash-bin99/branch-A-Assertions

Add assertions to the code

trash-bin99 committed:
Refactor and improve code quality

There is repeated code in the String related methods for the Task
subclasses.

This is unnecessary and harder to understand. It can be simplified.

Let's
* refactor the code such that it is simpler
* remove similar code

This will make the code easier to view and simpler to understand.

Other changes that can be made include:
* Remove unused method in the Ui class
* Add a constant for the minimum length of a input array
  in the Parser class

trash-bin99 committed:
Merge pull request #2 from trash-bin99/branch-A-CodeQuality

Refactor and improve code quality

trash-bin99 committed:
Add help command to the chatbot

Users do not have access to the commands usable by the chatbot.

This will cause users to be unable to use the full potential of the
chatbot.

Let's add a help command to the chatbot so that users have access to the
list of commands the chatbot accepts.

This will allow users to use the chatbot more easily.

trash-bin99 committed:
Add priority system to the chatbot

trash-bin99 committed:
Refactor code: Split classes into smaller packages

trash-bin99 committed:
Clean up import statements and add a new test

trash-bin99 committed:
Change pictures of the user and the chatbot

trash-bin99 committed:
Reallocate code into smaller methods

trash-bin99 committed:
Add screenshot to docs folder

trash-bin99 committed:
Edit README.md in docs

trash-bin99 committed:
Update README.md in docs

trash-bin99 committed:
Set theme jekyll-theme-architect
